### ISSUE TITLE
Failure to export tox file on first run of new qTox profile

### ISSUE THREAD

**Issue by truesilver92** [qTox/qTox#6301]
##### Brief Description

OS: Linux 5.10.7-3-MANJARO manjaro (aka arch) linux 
Manjaro package: qtox
Manjaro package version: 1.17.3-2
qTox version: v1.17.3
Commit hash:  c0e9a3b79609681e5b9f6bbf8f9a36cb1993dc5f
toxcore:  0.2.12
Qt:  5.15.2

Reproducible: Always

##### Steps to reproduce

1. Select My profile
2. Click `Export` in the `Profile` section
3. Attempt to save in home directory (or any other location the user has write access in)

##### Observed Behavior

Failure to save the tox file with an error message:
Title of window: `Failed to copy file`
Body of window: `The file you chose could not be written to.`

##### Expected Behavior

Export of tox file

##### Log

<details><summary>Contents of ~/.cache/qTox/qtox.log</summary>

```
[18:27:59.976 UTC] persistence/settings.cpp:117 : Debug: No settings file found, using defaults
[18:27:59.976 UTC] persistence/settings.cpp:121 : Debug: "Loading settings from :/conf/qtox.ini"
[18:28:00.028 UTC] ipc.cpp:61 : Debug: Our global IPC ID is  12050837600734416889
[18:28:00.028 UTC] main.cpp:312 : Debug: commit:  c0e9a3b79609681e5b9f6bbf8f9a36cb1993dc5f
[18:28:55.955 UTC] persistence/settings.cpp:2433 : Debug: Creating new profile settings in  "/home/joel/.config/tox//b4g.ini"
[18:28:55.959 UTC] persistence/settings.cpp:487 : Debug: Loading personal settings from "/home/joel/.config/tox/b4g.ini"
[18:28:55.959 UTC] core/toxoptions.cpp:110 : Debug: Core starting with IPv6 enabled
[18:28:56.004 UTC] persistence/db/rawdatabase.cpp:321 : Debug: Highest supported SQLCipher params on this system are "4.0 default"
[18:28:56.004 UTC] persistence/db/rawdatabase.cpp:292 : Debug: Setting SQLCipher "4.0 default" parameters
[18:28:56.004 UTC] persistence/db/rawdatabase.cpp:199 : Info: Opened database with SQLCipher "4.0 default" parameters
[18:28:56.125 UTC] persistence/history.cpp:240 : Debug: Database created at schema version 4
[18:28:56.126 UTC] nexus.cpp:86 : Debug: Starting up
[18:28:56.127 UTC] :0 : Warning: libpng warning: iCCP: known incorrect sRGB profile
[18:28:56.127 UTC] :0 : Warning: libpng warning: iCCP: cHRM chunk does not match sRGB
[18:28:56.136 UTC] platform/desktop_notifications/desktopnotify.cpp:34 : Debug: primary notification backend: "Freedesktop"
[18:28:56.365 UTC] persistence/profile.cpp:318 : Debug: Self avatar not found, will broadcast empty avatar to friends
[18:28:56.679 UTC] net/updatecheck.cpp:143 : Info: qTox is up to date
[18:28:57.840 UTC] core/core.cpp:484 : Debug: "Connecting to a bootstrap node..."
[18:28:57.840 UTC] core/core.cpp:484 : Debug: "Connecting to a bootstrap node..."
[18:28:58.641 UTC] network.c:556 : Warning: unknown address type: 0
[18:28:58.641 UTC] network.c:556 : Warning: unknown address type: 0
[18:28:59.641 UTC] network.c:556 : Warning: unknown address type: 0
[18:29:02.340 UTC] core/core.cpp:484 : Debug: "Connecting to a bootstrap node..."
[18:29:02.355 UTC] core/core.cpp:484 : Debug: "Connecting to a bootstrap node..."
[18:29:03.641 UTC] core/core.cpp:446 : Debug: Connected to the DHT
[18:31:24.651 UTC] audio/backend/openal.cpp:441 : Debug: Opening audio output "JDS Labs Element DAC Analog Stereo"
[18:31:24.652 UTC] audio/backend/openal.cpp:453 : Debug: Opened audio output "JDS Labs Element DAC Analog Stereo"
[18:31:24.653 UTC] audio/backend/openal.cpp:278 : Debug: Audio source 1 created. Sources active: 1
[18:31:24.655 UTC] audio/backend/openal.cpp:397 : Debug: Opening audio input "Elgato Wave:3 Mono"
[18:31:24.656 UTC] audio/backend/openal.cpp:423 : Debug: Opened audio input "Elgato Wave:3 Mono"
[18:31:24.656 UTC] audio/backend/openal.cpp:337 : Debug: Subscribed to audio input device [ 1 subscriptions ]
[18:31:24.656 UTC] video/camerasource.cpp:271 : Debug: Opening device "none" subscriptions: 1
[18:31:24.657 UTC] video/cameradevice.cpp:152 : Debug: Tried to open the null device
[18:31:24.657 UTC] video/camerasource.cpp:284 : Warning: Failed to open device!
[18:31:24.657 UTC] widget/form/settings/avform.cpp:393 : Debug: available Modes:
[18:31:24.657 UTC] widget/form/settings/avform.cpp:225 : Critical: Trying to select best mode from empty modes list
[18:31:24.657 UTC] widget/form/settings/avform.cpp:302 : Debug: selected Modes:
[18:31:45.860 UTC] audio/backend/openal.cpp:306 : Debug: Audio source 1 deleted. Sources active: 0
[18:31:45.860 UTC] audio/backend/openal.cpp:609 : Debug: Closing audio output
[18:31:45.860 UTC] audio/backend/openal.cpp:359 : Debug: Unsubscribed from audio input device [ 0 subscriptions left ]
[18:31:45.860 UTC] audio/backend/openal.cpp:583 : Debug: Closing audio input
[18:31:45.861 UTC] video/camerasource.cpp:388 : Debug: Closing device "none" subscriptions: 0
[18:35:27.601 UTC] video/camerasource.cpp:271 : Debug: Opening device "none" subscriptions: 1
[18:35:27.601 UTC] video/cameradevice.cpp:152 : Debug: Tried to open the null device
[18:35:27.601 UTC] widget/form/settings/avform.cpp:393 : Debug: available Modes:
[18:35:27.601 UTC] video/camerasource.cpp:284 : Warning: Failed to open device!
[18:35:27.601 UTC] widget/form/settings/avform.cpp:225 : Critical: Trying to select best mode from empty modes list
[18:35:27.601 UTC] widget/form/settings/avform.cpp:302 : Debug: selected Modes:
[18:35:27.601 UTC] audio/backend/openal.cpp:397 : Debug: Opening audio input "Elgato Wave:3 Mono"
[18:35:27.602 UTC] audio/backend/openal.cpp:423 : Debug: Opened audio input "Elgato Wave:3 Mono"
[18:35:27.602 UTC] audio/backend/openal.cpp:337 : Debug: Subscribed to audio input device [ 1 subscriptions ]
[18:35:27.602 UTC] audio/backend/openal.cpp:441 : Debug: Opening audio output "JDS Labs Element DAC Analog Stereo"
[18:35:27.603 UTC] audio/backend/openal.cpp:453 : Debug: Opened audio output "JDS Labs Element DAC Analog Stereo"
[18:35:27.604 UTC] audio/backend/openal.cpp:278 : Debug: Audio source 1 created. Sources active: 1
[18:35:30.270 UTC] audio/backend/openal.cpp:306 : Debug: Audio source 1 deleted. Sources active: 0
[18:35:30.270 UTC] audio/backend/openal.cpp:609 : Debug: Closing audio output
[18:35:30.270 UTC] audio/backend/openal.cpp:359 : Debug: Unsubscribed from audio input device [ 0 subscriptions left ]
[18:35:30.270 UTC] audio/backend/openal.cpp:583 : Debug: Closing audio input
[18:35:30.270 UTC] video/camerasource.cpp:388 : Debug: Closing device "none" subscriptions: 0
[18:35:41.489 UTC] video/camerasource.cpp:271 : Debug: Opening device "none" subscriptions: 1
[18:35:41.489 UTC] video/cameradevice.cpp:152 : Debug: Tried to open the null device
[18:35:41.489 UTC] widget/form/settings/avform.cpp:393 : Debug: available Modes:
[18:35:41.489 UTC] video/camerasource.cpp:284 : Warning: Failed to open device!
[18:35:41.489 UTC] widget/form/settings/avform.cpp:225 : Critical: Trying to select best mode from empty modes list
[18:35:41.489 UTC] widget/form/settings/avform.cpp:302 : Debug: selected Modes:
[18:35:41.489 UTC] audio/backend/openal.cpp:397 : Debug: Opening audio input "Elgato Wave:3 Mono"
[18:35:41.490 UTC] audio/backend/openal.cpp:423 : Debug: Opened audio input "Elgato Wave:3 Mono"
[18:35:41.490 UTC] audio/backend/openal.cpp:337 : Debug: Subscribed to audio input device [ 1 subscriptions ]
[18:35:41.490 UTC] audio/backend/openal.cpp:441 : Debug: Opening audio output "JDS Labs Element DAC Analog Stereo"
[18:35:41.490 UTC] audio/backend/openal.cpp:453 : Debug: Opened audio output "JDS Labs Element DAC Analog Stereo"
[18:35:41.491 UTC] audio/backend/openal.cpp:278 : Debug: Audio source 1 created. Sources active: 1
[18:35:42.478 UTC] audio/backend/openal.cpp:306 : Debug: Audio source 1 deleted. Sources active: 0
[18:35:42.478 UTC] audio/backend/openal.cpp:609 : Debug: Closing audio output
[18:35:42.478 UTC] audio/backend/openal.cpp:359 : Debug: Unsubscribed from audio input device [ 0 subscriptions left ]
[18:35:42.478 UTC] audio/backend/openal.cpp:583 : Debug: Closing audio input
[18:35:42.479 UTC] video/camerasource.cpp:388 : Debug: Closing device "none" subscriptions: 0
```

</details>

**Comment by anthonybilinski** [comment]
I can't repro this on tip myself on Ubuntu. The fact that you're getting that error indicates that there isn't already a file in the target location that you don't have write permission to, and that `QFile::copy` is just failing for some other reason.

I'm going to try running the Manjaro package in a VM to see if I can repro there, I wonder if there's some security policy maybe blocking writing.

**Comment by anthonybilinski** [comment]
Ok running Manjaro package version 1.17.3-2 as well, it fails on first run in the same way because the tox profile isn't written to disk yet until qTox is closed for the first time. After I close and re-open it once, exporting was correctly able to copy the file from `~/.config/tox/profile.tox` to `~/profile.tox`.

Do you happen to be trying to export the profile on first run?

**Comment by truesilver92** [comment]
Yes I was trying to export the profile on the first run. I'll try closing an exporting.

It worked

**Comment by anthonybilinski** [comment]
Ok, this still seems like not great behaviour for qTox. We could just save the tox profile to disk on first run right away instead of waiting until exit, so that export profile still works.

**Comment by anthonybilinski** [comment]
Oh actually this was fixed in 49c8f9b9b5dc3d364aa4ef13e22d199673ed68c3 already, which will be in qTox v1.18.0.
